### DISCUSSION TITLE
processorNonTransactional rollback plan

### DISCUSSION THREAD

**Discussion by 3jin-p** [spring-projects/spring-batch#4623] [Q&A]
Hi, I want to know the behavior of the transaction when using processorNonTransactional.
Specifying processorNonTransactional and trying to rollback the transactions in the chunk when exceptions occur within the processor?

Specify the connection timeout for db as 10 minutes,
In the processor that specifies processorNonTransactional
After 15 minutes, there was an exception

`"java.sql.SQLNonTransientConnectionException: Communications link failure during rollback(). Transaction resolution unknown.\n\tat com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:110)\n\tat com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)\n\tat com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)\n\tat com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)\n\tat com.mysql.cj.jdbc.ConnectionImpl.rollback(ConnectionImpl.java:1848)\n\tat com.zaxxer.hikari.pool.ProxyConnection.rollback(ProxyConnection.java:396)\n\tat com.zaxxer.hikari.pool.HikariProxyConnection.rollback(HikariProxyConnection.java)\n\tat org.springframework.jdbc.datasource.DataSourceTransactionManager.doRollback`

An error log like this occurred.
I'll change the connection timeout setting and take action
I'm so curious how it works

I'll be waiting for your reply. Thank you

**Comment by fmbenhassine** [comment]
If the processor is marked as non-transactional , then the results of item processing are cached across transactions in between retries and during skip processing, otherwise the processor will be called in every transaction.

I think your issue is not related to that. From what you shared, it seems that there is an attempt to access the database 5minutes after the timeout is elapsed (ie the is connection lost). Does this happen with a transactional processor? Again, I don't think the issue is related to that, but I just want to validate that.
